### PR TITLE
Detect potential secrets in envs or env_files

### LINKED ISSUE #136: Detect potential secrets in envs or env_files
As we'll need to handle local env_files at `init` and unpack them to kev configuration we might need to scan for common terms like PASSWORD, etc. and warn user when detected. 

User might refer to local env file for a set of environment variables and this could potentially contain secret info locally. Try and detect by scanning for commonly used terms and communicate to the user.

### PR DESCRIPTION
**Pull request by ezodude**
Resolves #136 

- [x] Found regexps to help match secrets from other OSS projects.
- [x] Retrofitted regexps as Secret Matchers to identify potential env var secret leaks.
- [x] Can detect secrets in manifest sources env vars.
- [x] Can detect secrets in environment override env vars.
- [x] Wired in the detect command to run after `init`.
- [x] Wired in the detect command to run after `Reconcile` and before `Render`.

### CODE REVIEW

**Review by marcinc — COMMENTED** [review]
Need a rebase

**Comment by ezodude** [comment]
@marcinc done (fyi used github directly).

File: pkg/kev/services.go [hunk 1]
```diff
@@ -66,6 +69,53 @@ func (s Services) Set() map[string]bool {
 	return out
 }
 
+func (s Services) detectSecrets(matchers []map[string]string, reporter io.Writer) error {
+	var matches []secretHit
+	for _, svc := range s {
+		matches = append(matches, svc.detectSecretsInEnvVars(matchers)...)
+	}
+
+	if nothingDetected := len(matches) < 1; nothingDetected {
```
Comment (marcinc) [inline pkg/kev/services.go:78]: why not just `if len(matches) == 0 { ... }` ?

File: pkg/kev/services.go [hunk 2]
```diff
@@ -66,6 +69,53 @@ func (s Services) Set() map[string]bool {
...
+				" → WARNING, leak detected    ... Service [%s], env var [%s], %s\n",
+				m.svcName,
+				m.envVar,
+				m.description)))
+	}
+	return nil
+}
+
+func (sc ServiceConfig) detectSecretsInEnvVars(matchers []map[string]string) []secretHit {
+	var out []secretHit
```
Comment (marcinc) [inline pkg/kev/services.go:95]: maybe call it `matches` like in the detectSecrets method above?

**Review by marcinc — APPROVED** [review]
2 tiny comments. LG otherwise.

File: pkg/kev/services.go (hunk 2 above)
Comment (marcinc) [reply pkg/kev/services.go:95]: or perhaps `hits`?

**Review by marcinc — APPROVED** [review]

**Pull request merged**